src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitter.java

```diff
@@ -0,0 +1,191 @@
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+@Scope(value="session")
+public class AssignmentSubmitter {
+    private static final Logger LOG = Logger.getLogger(AssignmentSubmitter.class);
+
+    @Autowired
+    private AttendanceAssignmentService assignmentService;
+
+    @Autowired
+    private AttendanceSectionService sectionService;
+
+    @Autowired
+    private AttendanceCourseService attendanceCourseService;
+
+    @Autowired
+    private AttendanceService attendanceService;
+
+    @Autowired
+    private CanvasApiWrapperService canvasApiWrapperService;
+
+    @Autowired
+    private AssignmentValidator assignmentValidator;
+
+    @Autowired
+    private CanvasAssignmentAssistant canvasAssignmentAssistant;
+
+
+    /**
+     * This function takes the summary for sections and one by one validates the attendance assignment and
+     * the canvas assignment. Then pushes all the grades and comments to the canvas assignment associated.
+     * @param isSimpleAttendance
+     * @param summaryForSections
+     */
+    public void submitCourseAttendances(boolean isSimpleAttendance, List<AttendanceSummaryModel> summaryForSections, Long courseId,
+                                        OauthToken oauthToken, AttendanceAssignment assignmentConfigurationFromSetup) throws AttendanceAssignmentException{
+
+        for (AttendanceSummaryModel sectionSummary : summaryForSections) {
+
+            AttendanceAssignment attendanceAssignment = assignmentService.findBySection(sectionService.getSectionInListById(courseId, sectionSummary.getSectionId()));
+
+            gradePushingValidation(courseId, oauthToken, assignmentConfigurationFromSetup, sectionSummary, attendanceAssignment);
+
+            submitSectionAttendances(isSimpleAttendance, sectionSummary, attendanceAssignment, courseId, oauthToken);
+        }
+    }
+
+    /**
+     * If in validation is determined discrepancy in the data between canvas and teh assignment in the data base,
+     * canvas will be aligned to the data in the db. If in validation is determined that there is not canvas assignment
+     * associated to the attendance assignment then a new canvas assignment will be created and associate to the attendance assignment.
+     */
+    private void gradePushingValidation(Long courseId, OauthToken oauthToken, AttendanceAssignment assignmentConfigurationFromSetup,
+                                        AttendanceSummaryModel model, AttendanceAssignment attendanceAssignment) throws AttendanceAssignmentException{
+
+            attendanceAssignment = assignmentValidator.validateConfigurationSetupExistence(model, attendanceAssignment);
+
+        if (attendanceAssignment.getStatus() == AttendanceAssignment.Status.UNKNOWN){
+            attendanceAssignment = assignmentValidator.validateAttendanceAssignment(courseId, attendanceAssignment, canvasApiWrapperService, oauthToken);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Introduce a new variable instead of reusing the parameter "attendanceAssignment". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1226)